library/src/main/java/com/google/android/exoplayer2/extractor/mkv/MatroskaExtractor.java

```diff
@@ -1608,4 +1628,118 @@ private static boolean parseMsAcmCodecPrivate(ParsableByteArray buffer) throws P
 
   }
 
+  private class TrueHDSampleHolder {
```

Comment:
Is there any other signaling in Matroska that indicates that a sample contains a TrueHD syncframe? Or is there a guarantee that 16 samples batched together will contain a syncframe at the start of one of the samples?

If so I think it might be possible to simplify/remove this. Calls to TrackOutput.sampleData and TrackOutput.sampleMetadata don't have to be matching in the sense that you can call sampleData a few times then call sampleMetadata once for the batch. So assuming you don't need to synchronize to a syncframe in the extractor you can just write data using the 'normal' readToOutput path then batch the samples using one call to sampleMetadata in commitSampleToOutput.

The platform AudioTrack should correctly synchronize to a syncframe internally (please let me know if this doesn't work for TrueHD) so the only remaining issue is that our AudioTrack wrapper needs to be able to read the syncframe header to get the size of the encoded sample in frames. Assuming a batch of 16 samples read in this extractor contains a syncframe, can we just move the code to find the syncframe header to the AudioTrack or Ac3Util so it only gets executed once after the flush? That seems preferable to introducing an extra copy here. Thanks!